src/main/java/com/es/phoneshop/model/entity/product/Product.java

```diff
@@ -36,6 +36,14 @@ public Product(String code, String description, BigDecimal price, Currency curre
         this.imageUrl = imageUrl;
     }
 
+    public void update(Product newProduct) {
```

Comment:
Please avoid any logic in entity classes. It's just a container for data.
Moreover let's imaghine that your Product has 100 attributes. In this case you method will be extremly big.
If you want to update product then just set this new product instead of a previous one.